File: services/product-service/app/src/main/java/io/micro_spring_boot/product_service/app/service/ProductService.java
package io.micro_spring_boot.product_service.app.service;

import io.micro_spring_boot.product_service.app.dto.CreateProductRequest;
import io.micro_spring_boot.product_service.app.dto.GetProductRequest;
import io.micro_spring_boot.product_service.app.dto.GetProductResponse;

import java.util.List;
import java.util.UUID;

public interface ProductService {
    void createProduct(CreateProductRequest productRequest);
    List<GetProductResponse> getAllProducts(GetProductRequest request);
    GetProductResponse getProductById(UUID id);
}

File: services/product-service/app/src/main/kotlin/com/microspringboot/productservice/app/services/ProductService.kt
package com.microspringboot.productservice.app.services

import com.microspringboot.productservice.app.dtos.CreateProductRequest
import com.microspringboot.productservice.app.dtos.GetProductRequest
import com.microspringboot.productservice.app.dtos.GetProductResponse
import java.util.*

interface ProductService {
    fun createProduct(productRequest: CreateProductRequest)
    fun getAllProducts(request: GetProductRequest): List<GetProductResponse>
    fun getProductById(id: UUID): GetProductResponse
}
